src/asynk/async_queue.rs

```diff
@@ -83,6 +86,54 @@ where
             None => Err(AsyncQueueError::TransactionEmpty),
         }
     }
+    pub async fn fetch_task(
```

Comment:
it's not implemented correctly. It should also update the task's state by changing it to "in_progress". Otherwise the same task will be processed by multiple workers
 
 fetching and updating should be done in the same transaction. 
 
 please read https://www.enterprisedb.com/blog/what-skip-locked-postgresql-95 to understand better what I mean